app/helpers/image_encode_helper.rb

```diff
@@ -0,0 +1,33 @@
+module ImageEncodeHelper
+  def encodable_image_tag(source, options = {})
+    image_tag(encodable_image_source(source), options)
+  end
+  alias eimage_tag encodable_image_tag
+
+  def encodable_image_source(source)
+    if base64_encode_images? && source.present?
+      base64_encoded_uri(source)
+    else
+      path_to_image(source)
+    end
+  end
+  alias eimage_source encodable_image_source
+
+  def base64_encode_images?
+    @base64_encode_images
+  end
+
+  def base64_encoded_uri(source)
+    asset = Rails.application.assets[source]
+
+    if asset.content_type == 'image/svg+xml'
+      encoding = 'charset=utf-8'
+      data = CGI.escape(asset.source).gsub('+', '%20')
```

Comment:
`ERB::Util.url_encode` probably? I know it does the same but in other cases we prefered the Active*/Rails variants over CGI